dashboard/app/controllers/levels_controller.rb

```diff
@@ -270,6 +273,21 @@ def update
     end
   end
 
+  # POST /levels/:id/update_start_code
+  # Update start code for a level. If params contain validation and start_sources,
+  # set those directly to ensure encryption. Otherwise set via update_properties
+  def update_start_code
+    if !@level.respond_to?(:start_sources) || !@level.respond_to?(:validation)
+      return update_properties
+    end
+    @level.start_sources = params[:start_sources]
+    @level.validation = params[:validation]
+    @level.log_changes(current_user)
+    @level.save!
```

Comment:
Are there any other properties we'd want to update? i.e. perhaps we could always do the call to `update_properties` but update `start_sources` and `validation` and first then perhaps edit the params so they don't have those fields.

Reply:
good idea, I'll try that out

Reply:
updated!